### ISSUE TITLE
Need to guard is_interactive better

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#3267]
While running benchmarks I noted that `time_ls*` ones were failing.
Running
```shell
asv run -j 4 --machine drogon-neurodebian10.20190322.1 -q -e -v -b api.supers.time_ls
```
helped to identify the reason:
```
                 File "/home/yoh/proj/datalad/datalad/venvs/benchmark-10.20190322.1/local/lib/python2.7/site-packages/asv/benchmark.py", line 569, in benchmark_timing
                   timing = timer.timeit(number)
                 File "/usr/lib/python2.7/timeit.py", line 202, in timeit
                   timing = self.inner(it, self.timer)
                 File "/usr/lib/python2.7/timeit.py", line 100, in inner
                   _func()
                 File "/home/yoh/proj/datalad/datalad/venvs/benchmark-10.20190322.1/local/lib/python2.7/site-packages/asv/benchmark.py", line 474, in <lambda>
                   func = lambda: self.func(*param)
                 File "/home/yoh/proj/datalad/datalad/benchmarks/api.py", line 126, in time_ls
                   ls(self.ds.path)
                 File "/home/yoh/proj/datalad/datalad/.asv/env/1b5862776063636a1bd6a1931ef20cce/local/lib/python2.7/site-packages/datalad/interface/ls.py", line 149, in __call__
                   return _ls_json(loc, json=json, **kw) if json else _ls_dataset(loc, **kw)
                 File "/home/yoh/proj/datalad/datalad/.asv/env/1b5862776063636a1bd6a1931ef20cce/local/lib/python2.7/site-packages/datalad/interface/ls.py", line 484, in _ls_dataset
                   path_fmt = u"{ds.path!U:<%d}" % (maxpath + (11 if is_interactive() else 0))  # + to accommodate ansi codes
                 File "/home/yoh/proj/datalad/datalad/.asv/env/1b5862776063636a1bd6a1931ef20cce/local/lib/python2.7/site-packages/datalad/utils.py", line 210, in is_interactive
                   return sys.stdin.isatty() and sys.stdout.isatty() and sys.stderr.isatty()
               ValueError: I/O operation on closed file
```
according to http://datalad.github.io/datalad/#api.supers.time_ls , the last commit which was successfully benchmarked was https://github.com/datalad/datalad/commit/3bcc58fa in Aug 8 2018... but that doesn't give me an immediate idea on what has happened since code for the is_interactive seemed to have no changes since then.

Will check in more detail later

**Comment by yarikoptic** [comment]
so the reason (not sure how changes yet caused it) but it is the stdin which is closed
```
               <closed file '<stdin>', mode 'r' at 0x7f39b73a20c0>
               <open file '<stdout>', mode 'w' at 0x7f39b73a2150>
               <open file '<stderr>', mode 'w' at 0x7f39b73a21e0>
```

**Comment by yarikoptic** [comment]
FWIW, I think it is something in `asv` which changed behavior which started to cause the failure, since previously passing benchmark state of datalad no longer does:
```shell
$ asv run -j 4 --machine drogon-neurodebian10.20190322.1 -q -e -b api.supers.time_ls  3bcc58f^^..3bcc58f^
...
[ 50.00%] ---- Finished setup for /tmp/tmpktBGFS
               Cleaning up /tmp/tmpktBGFS/testds10_
               Cleaning up /tmp/tmpktBGFS/testds10
               Traceback (most recent call last):
                 File "/home/yoh/proj/datalad/datalad/venvs/benchmark-10.20190322.1/local/lib/python2.7/site-packages/asv/benchmark.py", line 1039, in main_run_server
                   main_run(run_args)
                 File "/home/yoh/proj/datalad/datalad/venvs/benchmark-10.20190322.1/local/lib/python2.7/site-packages/asv/benchmark.py", line 913, in main_run
                   result = benchmark.do_run()
                 File "/home/yoh/proj/datalad/datalad/venvs/benchmark-10.20190322.1/local/lib/python2.7/site-packages/asv/benchmark.py", line 412, in do_run
                   return self.run(*self._current_params)
                 File "/home/yoh/proj/datalad/datalad/venvs/benchmark-10.20190322.1/local/lib/python2.7/site-packages/asv/benchmark.py", line 506, in run
                   min_run_count=self.min_run_count)
                 File "/home/yoh/proj/datalad/datalad/venvs/benchmark-10.20190322.1/local/lib/python2.7/site-packages/asv/benchmark.py", line 569, in benchmark_timing
                   timing = timer.timeit(number)
                 File "/usr/lib/python3.7/timeit.py", line 176, in timeit
                   timing = self.inner(it, self.timer)
                 File "<timeit-src>", line 6, in inner
                 File "/home/yoh/proj/datalad/datalad/venvs/benchmark-10.20190322.1/local/lib/python2.7/site-packages/asv/benchmark.py", line 474, in <lambda>
                   func = lambda: self.func(*param)
                 File "/home/yoh/proj/datalad/datalad/benchmarks/api.py", line 132, in time_ls_recursive_long_all
                   ls(self.ds.path, recursive=True, long_=True, all_=True)
                 File "/home/yoh/proj/datalad/datalad/.asv/env/a80b4bcff004e7d21aafa78a5932418b/lib/python3.7/site-packages/datalad/interface/ls.py", line 151, in __call__
                   return _ls_json(loc, json=json, **kw) if json else _ls_dataset(loc, **kw)
                 File "/home/yoh/proj/datalad/datalad/.asv/env/a80b4bcff004e7d21aafa78a5932418b/lib/python3.7/site-packages/datalad/interface/ls.py", line 483, in _ls_dataset
                   path_fmt = u"{ds.path!U:<%d}" % (maxpath + (11 if is_interactive() else 0))  # + to accommodate ansi codes
                 File "/home/yoh/proj/datalad/datalad/.asv/env/a80b4bcff004e7d21aafa78a5932418b/lib/python3.7/site-packages/datalad/utils.py", line 182, in is_interactive
                   return sys.stdin.isatty() and sys.stdout.isatty() and sys.stderr.isatty()
               ValueError: I/O operation on closed file
```

**Comment by kyleam** [comment]
Closed by #3268, which has made its way into master.  Not sure why this wasn't closed with that merge.
